presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java

```diff
@@ -285,6 +309,35 @@ public void addFinalQueryInfoListener(StateChangeListener<QueryInfo> stateChange
         stateMachine.addQueryInfoStateChangeListener(stateChangeListener);
     }
 
+    @Override
+    public Optional<QueryType> getQueryType()
+    {
+        if (statement instanceof Query) {
+            return Optional.of(SELECT);
+        }
+        else if (statement instanceof Explain) {
+            return Optional.of(EXPLAIN);
+        }
+        else if (statement instanceof ShowCatalogs  || statement instanceof ShowCreate || statement instanceof ShowFunctions ||
+                statement instanceof ShowGrants || statement instanceof ShowPartitions || statement instanceof ShowSchemas ||
+                statement instanceof ShowSession || statement instanceof ShowStats || statement instanceof ShowTables) {
+            return Optional.of(SHOW);
+        }
+        else if (statement instanceof CreateTableAsSelect) {
+            return Optional.of(CREATE);
```

Comment:
Maybe say 'CREATE_AS'? Otherwise, one would think of `CREATE`, which is actually classified as `DATA_DEFINITION`